Libraries/Aurora/Source/DirectX/PTRenderer.cpp

```diff
@@ -1460,17 +1465,23 @@ void PTRenderer::submitPostProcessing()
 
     // Set the root signature arguments for post-processing: the descriptor table (at the start of
     // the heap) and the post-processing settings. The descriptor table must start with the final
-    // texture. Then dispatch the post-processing shader, which tone maps (as needed) the
-    // accumulation texture (HDR) to the final texture (usually SDR).
-    // NOTE: Even if the settings mean no post-processing is performed, there is still an implicit
-    // format conversion, from the accumulation texture (UAV) format to the output texture format,
-    // e.g. floating-point to integer.
+    // texture.
     CD3DX12_GPU_DESCRIPTOR_HANDLE handle(_pDescriptorHeap->GetGPUDescriptorHandleForHeapStart(),
         kFinalDescriptorOffset, _handleIncrementSize);
     pCommandList->SetComputeRootDescriptorTable(0, handle);
     pCommandList->SetComputeRoot32BitConstants(
         1, sizeof(PostProcessing) / 4, &_postProcessingData, 0);
-    pCommandList->Dispatch(_outputDimensions.x, _outputDimensions.y, 1);
+
+    // Dispatch the post-processing shader, which tone maps(as needed) the accumulation texture
+    // (HDR) to the final texture (usually SDR).
+    // NOTE: This should be done even if the settings mean no post-processing is performed as there
+    // is still an implicit format conversion, from the accumulation texture (UAV) format to the
+    // output texture format, e.g. floating-point to integer.
+    // NOTE: The dispatch is performed with thread group dimensions that provide good occupancy for
+    // the current compute shader code. This must match the values in the compute shader.
+    constexpr uvec2 kThreadGroupCount(16, 8);
```

Comment:
Note we could compute this from WaveLaneCountMin in D3D12_FEATURE_DATA_D3D12_OPTIONS1 , though we'd need to runtime compile the shader to do that.
https://learn.microsoft.com/en-us/windows/win32/api/d3d12/ns-d3d12-d3d12_feature_data_d3d12_options1

Reply:
Using `WaveLaneCountMin` here probably won't give the best performance -- `WaveLaneCountMin` is 32 on NVIDIA GPUs and 32 or 64 on other devices. For this shader, using 32 threads per thread group would give an occupancy of 1/3 on compute capability 8.6 GPUs according to the CUDA Occupancy Calculator. Here, 16*8 == 128 threads works well because you want to make sure each SM always has its maximum number of threads running if possible; this works out to at least 64 threads per thread group on Turing, and 96 threads per thread group on Ampere. I've sent some more information on the math behind how this works out, but the short explanation is that 128, 256, or 512-thread groups work well on every architecture from Kepler to the present day.

Getting the number of threads per SM is a bit tricky; I'm not sure if it's a feature in DirectX, but it can be queried from CUDA; it's `cudaDeviceProp::maxThreadsPerMultiProcessor`. That times 32 divided by `cudaDeviceProp::maxBlocksPerMultiprocessor` will give you the minimum number of threads per block that should be used (64 on Turing, 96 on Ampere), which could then be factorized to give these thread group dimensions.